chia/farmer/farmer.py

```diff
@@ -67,9 +67,10 @@
 
 
 class HarvesterCacheEntry:
-    def __init__(self):
+    def __init__(self, update_interval: int):
         self.data: Optional[dict] = None
         self.last_update: float = 0
+        self.update_interval = update_interval
```

Comment:
it seems a bit odd to store this interval in *every* cache entry. It's going to be the same value in all of them, right?

You could just pass it into `needs_update()` and `expired()` instead.

Alternatively, you could pass it into `bump_last_update()` and change the `last_update` timestamp to instead be *next_update* timestamp.

Reply:
Agreed, having it stored in each cache entry is weird. See f80e912b1d8cc17a576d8d8e0637dfe2e65e0ac7.